airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_stream.py

```diff
@@ -105,15 +106,29 @@ def cursor_field(self) -> Union[str, List[str]]:
         """
         return self.stream_cursor_field
 
+    def read_records_as_messages(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[AirbyteMessage]:
+        # FIXME: safeguard this assignment!
+        if hasattr(self.logger, "level"):
+            self.retriever.logger.setLevel(self.logger.level)
+        for message in self.retriever.read_records_as_message(sync_mode, cursor_field, stream_slice, stream_state):
+            if message.type == MessageType.RECORD:
+                message.record.data = self._apply_transformations(message.record.data, self.config, stream_slice)
+            yield message
+
     def read_records(
         self,
         sync_mode: SyncMode,
         cursor_field: List[str] = None,
         stream_slice: Mapping[str, Any] = None,
         stream_state: Mapping[str, Any] = None,
     ) -> Iterable[Mapping[str, Any]]:
-        for record in self.retriever.read_records(sync_mode, cursor_field, stream_slice, stream_state):
-            yield self._apply_transformations(record, self.config, stream_slice)
+        raise RuntimeError("deprecated")
```

Comment:
this method is still part of the Stream interface, but AbstractSource calls `read_records_as_messages` instead